### PR TITLE
feat: New UI for SignUp / Login

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4365, 4366
- **Design**: Checked with Ciaran ✅
- **Logic updates**: 
1. Decouple google sign-in from email sign-up form values (`disabled` prop or `searchQueryParams` that is added to Google auth url no longer depend on sign-up form values)
2. When signup URL link has `token` param, display the email form w/o a back button
3. Email form will have a button saying "Create Account with SAML" if user clicked the "SAML" button rather than "Continue with email"
4.  "Create Account with SAML" is disabled if email / username isn't there
5. Remove password field for SAML signup

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Test /signup and /auth/login

### CODE REVIEW

File: apps/web/modules/signup-view.tsx [hunk 1]
```diff
@@ -333,6 +336,19 @@ export default function Signup({
           <HeadSeo title={t("sign_up")} description={t("sign_up")} />
           {/* Left side */}
           <div className="ml-auto mr-auto mt-0 flex w-full max-w-xl flex-col px-4 pt-6 sm:px-16 md:px-20 lg:mt-12 2xl:px-28">
+            {displayBackButton && (
+              <div className="flex w-fit lg:-mt-12">
+                <Button
+                  color="minimal"
+                  className="hover:bg-subtle todesktop:mt-10 [&[aria-current='page']]:bg-emphasis [&[aria-current='page']]:text-emphasis group-hover:text-default text-emphasis group mb-6 flex h-6 max-h-6 w-full flex-row items-center rounded-md px-3 py-2 text-sm font-medium leading-4 transition"
```
Comment (joeauyeung) [inline apps/web/modules/signup-view.tsx:343]: I see your using the imported `<Button />` component from the UI library. Why are there so many classNames here? Wouldn't this button also match the rest of the buttons we use through out the app?

**Review by joeauyeung — CHANGES_REQUESTED** [review]
Great work so far @hbjORbj I just have some blocking comments

## Blocking
- I'm able to sign up without consenting to the privacy statement or cookies
- Likewise when submitting a premium username and returning a server error I'm still redirected to the confirmation email page

## Non-blocking
- Icons are still messed up for me but does this button look aligned when the icon renders?
image

File: apps/web/modules/signup-view.tsx (hunk 1 above)
Comment (hbjORbj) [reply apps/web/modules/signup-view.tsx:343]: removed unneeded classnames!

**Comment by hbjORbj** [comment]
> Icons are still messed up for me but does this button look aligned when the icon renders?
Can you show me what icons are not as expected?

**Comment by hbjORbj** [comment]
> I'm able to sign up without consenting to the privacy statement or cookies

Good point! But this was already this way prior to this PR. I will ask around if this is designed or a big.

**Review by joeauyeung — CHANGES_REQUESTED** [review]
Talking to @hbjORbj about this but we found a bug with signing up with SAML.

**Comment by hbjORbj** [comment]
In our current logic, SAML button is coupled with the email form. They must be on the same page, or we need to change the logic. Talking with Ciaran now

File: apps/web/modules/signup-view.tsx [hunk 2]
```diff
@@ -531,13 +386,207 @@ export default function Signup({
...
+                    <Alert
+                      className="mb-3"
+                      severity="error"
+                      message={errors.apiError?.message}
+                      data-testid="signup-error-message"
+                    />
+                  )}
+                  {isSamlSignup ? (
+                    <Button
+                      type="submit"
```
Comment (hbjORbj) [inline apps/web/modules/signup-view.tsx:464]: added

File: apps/web/modules/signup-view.tsx [hunk 3]
```diff
@@ -531,13 +386,207 @@ export default function Signup({
...
+                      color="primary"
+                      disabled={
+                        !!formMethods.formState.errors.username ||
+                        !!formMethods.formState.errors.email ||
+                        !formMethods.getValues("email") ||
```
Comment (hbjORbj) [inline apps/web/modules/signup-view.tsx:469]: added

File: apps/web/modules/signup-view.tsx [hunk 4]
```diff
@@ -347,177 +364,15 @@ export default function Signup({
...
-                      if (!formMethods.getValues("username")) {
-                        formMethods.trigger("username");
-                      }
-                      if (!formMethods.getValues("email")) {
-                        formMethods.trigger("email");
 
-                        return;
-                      }
+            {/* Form Container */}
+            {displayEmailForm && (
```
Comment (hbjORbj) [inline apps/web/modules/signup-view.tsx:369]: flag

File: apps/web/modules/signup-view.tsx [hunk 5]
```diff
@@ -347,177 +364,15 @@ export default function Signup({
...
+              <div className="mt-12">
+                <Form
+                  className="flex flex-col gap-4"
+                  form={formMethods}
+                  handleSubmit={async (values) => {
+                    if (isSamlSignup) {
```
Comment (hbjORbj) [inline apps/web/modules/signup-view.tsx:375]: saml button's onClick logic is moved to here

File: apps/web/modules/signup-view.tsx [hunk 6]
```diff
@@ -531,13 +386,208 @@ export default function Signup({
...
+                  )}
+                  {isSamlSignup ? (
+                    <Button
+                      type="submit"
+                      color="primary"
+                      disabled={
+                        !!formMethods.formState.errors.username ||
+                        !!formMethods.formState.errors.email ||
+                        !formMethods.getValues("email") ||
+                        !formMethods.getValues("username") ||
```
Comment (hbjORbj) [inline apps/web/modules/signup-view.tsx:470]: added

**Review by joeauyeung — APPROVED** [review]

**Pull request merged**